### PR TITLE
Update rendermime docs

### LINKED ISSUE #772: Improve docs for RenderMime.IRenderer
It is not clear when isSafe and isSanitizable should return true and when false. Please explain assumptions about these methods and what sanitization and sanitized mean.

### PR DESCRIPTION
**Pull request by blink1073**
Fixes #772.

### CODE REVIEW

File: src/rendermime/index.ts [hunk 1]
```diff
@@ -225,11 +225,20 @@ namespace RenderMime {
 
     /**
      * Whether the input is safe without sanitization.
+     *
+     * #### Notes
+     * A `safe` output is one that cannot pose a security threat
+     * when added to the DOM, for example when it is set as `.textContent`.
```
Comment (jasongrout) [inline src/rendermime/index.ts]: for example when text is added with `.textContent`.

**Comment by jasongrout** [comment]
I think we should also add a few sentences in the Rendermime class-level docs explaining the rendering process, and in particular, when a sanitizer is looked for/called (and who calls it), or when isSafe is evaluated.

**Comment by ellisonbg** [comment]
Looks good, I will let @jasongrout merge as he had the comment above.

File: src/rendermime/index.ts [hunk 2]
```diff
@@ -108,8 +118,9 @@ class RenderMime {
    * @param trusted - whether the bundle is trusted.
    *
    * #### Notes
-   * If the bundle is not trusted, the highest preference
-   * mimetype that is sanitizable or safe will be chosen.
+   * For untrusted bundles, only select mimetypes that can be rendered
+   * "safely"  (see [[RenderMime.IRenderer.isSafe]]) or can  be "santized"
```
Comment (jasongrout) [inline src/rendermime/index.ts]: san**i**tized

File: src/rendermime/index.ts [hunk 3]
```diff
@@ -225,11 +236,22 @@ namespace RenderMime {
...
+     * when added to the DOM, for example when text is added with
+     * `.textContent`.
      */
     isSafe(mimetype: string): boolean;
 
     /**
      * Whether the input can safely sanitized for a given mimetype.
+     *
+     * #### Notes
+     * A `santizable` output is one that could pose a security threat
```
Comment (jasongrout) [inline src/rendermime/index.ts]: san`i`tizable

File: src/rendermime/index.ts [hunk 4]
```diff
@@ -35,11 +35,16 @@ interface IRenderMime extends RenderMime {}
...
+ * bundle determines the renderer that will be used.
  *
- * You can add a renderer by adding it to the `renderers` object and registering
- * the mimetype in the `order` array.
+ * You can add a renderer by adding it to the `renderers` object and
+ * registering the mimetype in the `order` array.
+ *
+ * Untrusted bundles are handled differently from trusted ones.  Untrusted
+ * bundles will only render outputs that can be rendered "safely"
+ * (see [[RenderMime.IRenderer.isSafe]]) or can be "santized"
```
Comment (jasongrout) [inline src/rendermime/index.ts]: san`i`tized

**Comment by jasongrout** [comment]
Looks good, module some typos noted inline.

**Pull request merged**